### ISSUE TITLE
uri module reports The server's certificate does not match with its hostname (wildcard!)

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible#14979]
##### Issue Type:
- Bug Report
  ##### Ansible Version:

```
ansible 2.0.1.0
  config file = /Users/sorins/bmll/devops/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Summary:

Ansible reports "The server's certificate does not match with its hostname." even when the certificate is perfectly valid. 

We do have a wildcard certificate on the entire domain, but ansible seems to fail to validate the certificate.
##### Steps To Reproduce:

Perform a URI request against a webserver that has a wildcard certificate on it. Any browser will display the certificate as valid, but ansible will fail to validate it.

**Comment by abadger** [comment]
This looks like it has been fixed on the devel branch (will be 2.1.0) with the change to uri to use our own fetch_url() functions instead of httplib2.  This change was backported to stable-2.0 in time to make 2.0.2 as well.  Please try out 2.0.2.0-rc1 from http://releases.ansible.com/ansible/ to make sure that the bug has been addressed (and report any new bugs you encounter).

**Comment by abadger** [comment]
2.0.2.0 has been released so I'm going to close this.
